### ISSUE TITLE
Simplify how we configure New Relic agent in Heroku.

### ISSUE THREAD

**Issue by DFurnes** [DoSomething/infrastructure#60]
We've historically configured New Relic on our Heroku apps by attaching a "placeholder" [free-tier New Relic add-on](https://elements.heroku.com/addons/newrelic) and then replacing the auto-generated `NEW_RELIC_LICENSE_KEY` environment variable with our own. While this works, it results in some inconsistent behavior –

1. New Relic _must_ be installed on QA environments, since that's where builds happen. If the application that a build is promoted from doesn't have the add-on, New Relic won't be installed. (And moving forward, we likely won't want New Relic on QA environments to reduce usage!)
2. Each app gets a standalone unused New Relic account, which has led to confusion.

When setting this up [for Longshot](https://github.com/DoSomething/infrastructure/issues/49#issuecomment-442632020), I was able to simplify setup to make this more reliable:

> For future reference, the steps followed:
> 1. Add the `ext-newrelic` PHP extension to the application, instead of using a Heroku addon. This has the advantage of not clobbering environment variables & provisioning a parallel "mini-account" in Heroku that goes unused. (DoSomething/longshot#923)
> 2. Add support for per-environment `newrelic.enabled`, so we can install the New Relic agent in the initial Heroku build but only enable it when the same compiled slug is promoted to a production app.  (DoSomething/longshot#924)
> 3. Set the proper environment variables per-application, via `with_newrelic` arg. (#59)

If we continue using New Relic, we should take this approach in other applications as well.

**Comment by mshmsh5000** [comment]
+1 on the general approach, and on finding a way not to have QA instances reporting in to New Relic's APM.

Is there a parallel approach for Node/npm?

**Comment by DFurnes** [comment]
> Is there a parallel approach for Node/npm?

Yep, and I think we're already doing that! We've [installed the `newrelic` NPM package](https://docs.newrelic.com/docs/agents/nodejs-agent/installation-configuration/install-nodejs-agent) and configured it ourselves in Gambit & Blink, so these should not have the same issues we'd faced in PHP-land.

**Comment by mshmsh5000** [comment]
Spicy.

**Comment by DFurnes** [comment]
I've configured environment variables for this in Northstar, Phoenix, and Rogue in #87. Next step is to add the `ext-newrelic` extension to each application's `composer.json` and remove the add-on.

**Comment by DFurnes** [comment]
Oh, since our Terraform config defaults the New Relic application name to the application name (so for example, `dosomething-northstar` rather than `Northstar Production Heroku` like we'd had before), these will show up as new apps. 😯

I've updated our alerting policies to refer to these new applications, but history unfortunately can't transfer over. The old ones have been automatically marked as inactive, so I'll remove those from our account so they hopefully won't count as new hosts in our billing. 🤞 

cc: @mshmsh5000

**Comment by mshmsh5000** [comment]
We're almost definitely switching to their compute-based model, which doesn't keep track of individual applications or hosts anymore. So this shouldn't matter.

**Comment by DFurnes** [comment]
Phoenix, Northstar, and Rogue are all configured with `ext-newrelic` in their dependencies (and so no longer need the add-on installed on QA to "force" that for the builds we promote to production), and with #87 we can enable/disable this extension via Terraform without any redeploys.

I've removed the New Relic Heroku extensions from each of these applications, and will keep an eye out that we continue to get metrics over the next couple of days. Closing! 🔨
